### PR TITLE
Implement `instant_print_lines` argument

### PR DESCRIPTION
**Pull request by josherrickson**
Addresses #582; adding functionality to remove any time-delay in printing the first n number of lines.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -16,6 +16,7 @@ use modules::{Module, ALL_MODULES};
 lazy_static::lazy_static! {
     pub static ref CTRLC_PRESSED: AtomicBool = AtomicBool::new(false);
     pub static ref SPEED_FACTOR: Mutex<f32> = Mutex::new(1.0);
+    pub static ref INSTANT_PRINT_LINES: Mutex<u32> = Mutex::new(0);
```
Comment (svenstaro) [inline src/lib.rs:19]: Can't you use `AtomicU32` here instead?

File: src/io.rs [hunk 2]
```diff
@@ -6,13 +6,24 @@ use wasm_bindgen::prelude::*;
...
     let speed_factor = *SPEED_FACTOR.lock().await;
-    let sleep_length = time::Duration::from_millis((1.0 / speed_factor * length as f32) as u64);
+    let count = COUNTER.fetch_add(1, Ordering::Relaxed);
+    let sleep_length = if count < *INSTANT_PRINT_LINES.lock().await {
+        // If user passed `--instant-print-lines`, there should be
+        // no pauses in first `INSTANT_PRINT_LINES` number of lines
+        time::Duration::new(0, 0)
+    } else {
+        time::Duration::from_millis((1.0 / speed_factor * length as f32) as u64)
+    };
```
Comment (svenstaro) [inline src/io.rs:26]: Please also handle the `wasm32` case.

File: src/args.rs [hunk 3]
```diff
@@ -37,6 +43,10 @@ pub struct AppConfig {
     #[clap(short, long, default_value = "1", value_parser = parse_speed_factor)]
     pub speed_factor: f32,
 
+    /// Instantly print this many lines
+    #[clap(short, long = "instant-print-lines", default_value = "0", value_parser = parse_instant_print)]
```
Comment (svenstaro) [inline src/args.rs:47]: Please also implement this for `wasm32`.

**Review by svenstaro — CHANGES_REQUESTED** [review]
Great first contribution! A few nitpicks but the overall concept looks sound!

File: src/args.rs [hunk 4]
```diff
@@ -12,6 +12,12 @@ fn parse_speed_factor(s: &str) -> Result<f32, String> {
     Ok(value_as_float)
 }
 
+#[cfg(not(target_arch = "wasm32"))]
+fn parse_instant_print(s: &str) -> Result<u32, String> {
+    let value_as_int = s.parse::<u32>().map_err(|e| e.to_string())?;
+    Ok(value_as_int)
+}
```
Comment (svenstaro) [inline src/args.rs:19]: Do you actually need this? If you just specify `u32`, it should already parse this correctly I think.

File: src/lib.rs (hunk 1 above)
Comment (josherrickson) [reply src/lib.rs:19]: When I switch to `AtomicU32`, the value of `INSTANT_PRINT_LINES.load(Ordering::CstSeq)` in io.rs is always 0 regardless of whether I pass in the flag. Any suggestions of where to start looking to address this?

File: src/io.rs (hunk 2 above)
Comment (svenstaro) [reply src/io.rs:26]: Yes, `make web` and then launch a web server on `dist/`. You will have to install `trunk`: `cargo install trunk`. An easy web server you can also install with cargo: `cargo install miniserve`. Then do `miniserve dist/` and then just go there with your browser.

File: src/lib.rs (hunk 1 above)
Comment (josherrickson) [reply src/lib.rs]: Nevermind, got it. Will push a fix up shortly.

File: src/io.rs (hunk 2 above)
Comment (josherrickson) [reply src/io.rs:26]: I'm unable to get the web server running, either on my local macbook, or a remote RHL server. Both my fork, and a clone from your local version. It gives 404 for the .wasm and .css files.

Comment (svenstaro) [reply src/io.rs:26]: Try this: `miniserve dist/ --route-prefix genact --spa --index index.html`

**Comment by svenstaro** [comment]
Could you also rebase to get rid of the merge commit into your branch?

**Comment by josherrickson** [comment]
I've added wasm32 support. I also rebased.

PS I picked `instant-print-lines` just to avoid `-q`, but I suppose `-i` isn't much better - let me know if you'd prefer I use `quickstart` or something else.

**Comment by svenstaro** [comment]
Great work! Not bad for your first Rust contribution. :)

**Pull request merged**